src/components/UserProfile/BasicInformationTab/__tests__/BasicInformationTab.test.js

```diff
@@ -0,0 +1,600 @@
+
+import {Name, Title, Email, formatPhoneNumber , Phone, TimeZoneDifference} from '../BasicInformationTab';
+import BasicInformationTab from '../BasicInformationTab';
+import { render, screen, fireEvent,   waitFor, act } from '@testing-library/react';
+import { toast } from 'react-toastify';
+import thunk from 'redux-thunk';
+import configureStore from 'redux-mock-store';
+import { Provider } from 'react-redux';
+import mockAdminState from '__tests__/mockAdminState';
+
+
+// Mock components 
+jest.mock('../../UserProfileEdit/ToggleSwitch', () => (props) => <div data-testid="toggleswitch">{props.id}</div>);
+jest.mock('react-phone-input-2', () => (props) => <div data-testid="phoneinput">{props.id}</div>);
+jest.mock('../../EditableModal/EditableInfoModal', () => () => <div data-testid="editablemodel">RoleInfo</div>);
+
+
+//Test Suites 
+describe('Test Suite for Name component', () => {
+   let testProps= {
+    userProfile: { firstName: 'John', lastName: 'Doe' },
+    setUserProfile: jest.fn(),
+    formValid: { firstName: true, lastName: true },
+    setFormValid: jest.fn(),
+    canEdit: true,
+    desktopDisplay: 3,
+  };
```

Comment:
You repeat setup code such as testProps across different test suites. Consider using a global beforeEach to define common props and reset mocks to avoid redundancy and make tests easier to manage.

```suggestion
  describe('Test Suite for Name component', () => {
  let testProps;

  beforeEach(() => {
    testProps = {
      userProfile: { firstName: 'John', lastName: 'Doe' },
      setUserProfile: jest.fn(),
      formValid: { firstName: true, lastName: true },
      setFormValid: jest.fn(),
      canEdit: true,
      desktopDisplay: 3,
    };
    jest.clearAllMocks();
  });

  it('Test case 1: Name component renders with editable fields when canEdit is true', () => {
    render(<Name {...testProps} />);
    const firstNameField = screen.getByTestId('firstName');
    expect(firstNameField).toBeInTheDocument();
    expect(firstNameField.value).toEqual('John');
    const lastNameField = screen.getByTestId('lastName');
    expect(lastNameField).toBeInTheDocument();
    expect(lastNameField.value).toEqual('Doe');
  });


});
```